### PR TITLE
Add theme-context to button and card

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary
This pr updates terra-button and terra-card to use the new theme-context to apply themes.

### Additional Details

Thanks for contributing to Terra.
@cerner/terra

[CONTRIBUTORS.md]: ../CONTRIBUTORS.md
[License]: ../LICENSE

### CODE REVIEW

File: packages/terra-button/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ ChangeLog
 
 Unreleased
 ----------
+### Changed
+* Directly include packaged themes
```
Comment (emilyrohrbough) [inline packages/terra-button/CHANGELOG.md:7]: should we add consume terra-theme-context to these?

Comment (mjhenkes) [reply packages/terra-button/CHANGELOG.md:7]: Yeah I need to update those.

File: packages/terra-button/tests/jest/Button.test.jsx [hunk 2]
```diff
@@ -225,4 +227,13 @@ it('should render a button with accessible label', () => {
...
 });
+
+it('correctly applies the theme context className', () => {
+  const button = mount(
+    <ThemeContextProvider theme={{ className: 'orion-fusion-theme' }}>
+      <Button text="neutral" />
+    </ThemeContextProvider>,
+  );
+  expect(button).toMatchSnapshot();
+});
```
Comment (mjhenkes) [inline packages/terra-button/tests/jest/Button.test.jsx:238]: Just a note here: I didn't like how I ended up having to write this test. The context api's for enzyme did not work, mocking was taking too long to figure out and shallow rendering still did not apply the context. 
I don't want consumers pulling in the terra-theme-context package directly to get access to the provider... If people have to test this way, we should provide the provider from the terra-application package.

File: .travis.yml [hunk 3]
```diff
@@ -35,7 +35,7 @@ jobs:
     - stage: danger lint and jest
       script: travis_wait docker-compose run -e DANGER_GITHUB_API_TOKEN=$DANGER_GITHUB_API_TOKEN -e HAS_JOSH_K_SEAL_OF_APPROVAL=$HAS_JOSH_K_SEAL_OF_APPROVAL -e TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST -e TRAVIS_REPO_SLUG=$TRAVIS_REPO_SLUG test-ci sh -c 'npm run danger && npm run lint && npm run jest'
     - stage: compile - default theme
-      script: travis_wait docker-compose run test-ci npm run compile:prod
+      script: travis_wait docker-compose run test-ci npm run compile:prod -- --env.disableAggregateThemes
```
Comment (mjhenkes) [inline .travis.yml:38]: Performance boost to not aggregate themes while testing the default.

File: packages/terra-button/tests/jest/Button.test.jsx (hunk 2 above)
Comment (emilyrohrbough) [reply packages/terra-button/tests/jest/Button.test.jsx:238]: I'm pretty sure the enzyme context API only works when you set context like:
```
Component.context = Context;
```
I had my own slew of troubles getting the mocking to work for terra-action-header and opted to use `React.useContext` so I could mock like:
```
jest.spyOn(React, 'useContext).mockReturnValue({ className: 'orion-fusion-theme' });
```

Comment (StephenEsser) [reply packages/terra-button/tests/jest/Button.test.jsx:238]: +1 to exposing a theme provider in terra-application

File: packages/terra-button/src/terra-dev-site/test/button/ActiveVariantButtons.test.jsx [hunk 4]
```diff
@@ -1,61 +1,63 @@
...
 import Button from '../../../Button';
 import styles from './ButtonTestCommon.module.scss';
 
 const cx = classNames.bind(styles);
+const cy = classNames.bind(buttonStyles);
 
 const NeutralButton = () => (
   <div id="neutral" className={cx('spacing')}>
-    <Button id="neutralActive" text="Neutral-Active" variant="neutral" className="is-active" />
+    <Button id="neutralActive" text="Neutral-Active" variant="neutral" className={cy('is-active')} />
```
Comment (StephenEsser) [inline packages/terra-button/src/terra-dev-site/test/button/ActiveVariantButtons.test.jsx:13]: Fixed the class names binding loophole, broke the tests

image

Comment (StephenEsser) [reply packages/terra-button/src/terra-dev-site/test/button/ActiveVariantButtons.test.jsx:13]: For a temporary workaround to move this PR forward I think this makes sense, in the long run we might consider flagging this to update later. Either find a way to press the active state in a test or find another way to test the functionality. I don't think we should rely on local class names outside of the class for testing.

File: packages/terra-button/tests/jest/Button.test.jsx (hunk 2 above)
Comment (tbiethman) [reply packages/terra-button/tests/jest/Button.test.jsx:238]: Enzyme's context APIs for shallow/mount use the old context API, which we're avoiding completely. This is annoying here, but the uplift to react-intl v4 is going to be much more disruptive (no more shallow/mountWithIntl :S).

**Pull request merged**